tests/system/src/test/java/test/util/Util.java

```diff
@@ -63,21 +63,8 @@ private static interface Future {
         public abstract boolean await(long timeout, TimeUnit unit);
     }
 
-    public static void throwError(Throwable testError) {
-        if (testError != null) {
-            if (testError instanceof Error) {
-                throw (Error)testError;
-            } else if (testError instanceof RuntimeException) {
-                throw (RuntimeException)testError;
-            } else {
-                AssertionFailedError err = new AssertionFailedError("Unknown exception");
-                err.initCause(testError.getCause());
-                throw err;
-            }
-        } else {
-            AssertionFailedError err = new AssertionFailedError("Unexpected exception");
-            throw err;
-        }
+    public static void throwError(Throwable e) {
+        fail(e);
     }
```

Comment:
This is not an equivalent change. In the former code, if `testError` was a RuntimeException it was thrown directly and not wrapped. I recommend reverting this change, and then replacing only the two blocks that wrap the throwable in an AssertionError with `fail(testError)`.